### ISSUE TITLE
Provider plugins doesn't show the full name of ISP

### ISSUE THREAD

**Issue by tomac4t** [matomo-org/matomo#15674]
- Matomo version: 3.13.12

I notice that Provider plugins don't show the full name of ISP: 

![2020-03-06_00-56](https://user-images.githubusercontent.com/47532693/76005490-0e456600-5f46-11ea-91f6-dbaf9a11adcc.png)

The example IP is 113.122.0.0, here is its information: 

```bash
$ mmdblookup -f GeoLite2-ASN.mmdb -i 113.122.0.0

  {
    "autonomous_system_number": 
      4134 <uint32>
    "autonomous_system_organization": 
      "No.31,Jin-rong Street" <utf8_string>
  }
```
Its full name is "No.31,Jin-rong Street" instead of "No" in GeoLite2-ASN. (Actually, it is an address rather than organization name... Maxmind error data.)

The related code is here, 

https://github.com/matomo-org/matomo/blob/45dd3a9e5c6513f48ed55b51ecda185fb4bdf837/plugins/Provider/functions.php#L22-L31

I don't know the reason why `substr` part is existing, maybe it could be removed safely?

**Comment by tsteur** [comment]
@tomac4t good finding of where things go wrong in that case. I think it's actually meant to behave like this and maybe only an issue because MaxMind reports the address here instead of the actual provider name?

**Comment by tomac4t** [comment]
@tsteur Yes, I download their ASN CSV files and search "No." text in there. They are take so many address as its provider name: 
```
1.1.8.0/24,4134,"No.31,Jin-rong Street"
1.71.144.0/20,132147,"No.3,Shu-Ma Road"
45.250.43.0/24,58461,"No.288,Fu-chun Road"
103.10.9.0/24,56285,"Plot No.770, Road No.44"
103.10.222.0/24,38275,"5th Floor, Plot No. 39, Phase II, Block A"
103.12.144.0/22,132326,"K-7-10, No. 2 Jalan Solaris, Solaris Mon't Kiara"
103.20.125.0/24,132417,"Plot No. 5& 43, HITECH City, Madhapur"
103.20.126.0/23,132415,"Palm Court Bldg M, 501/B, 5th Floor, New Link Road"
103.21.18.0/23,58714,"19th Floor, Tower C and D, Building No. 14, DLF Cyber City"
103.21.44.0/22,132435,"No.31-A, Jalan Tiara, Tiara Square"
103.22.156.0/22,45504,"4F-1 NO. 20 Ta-Lung Road"
103.28.140.0/22,132108,"IL&FS House, Plot No. 14"
103.94.16.0/22,135916,"descr: No. 14, 256 Bach Dang, Chuong Duong, Hoan Kiem, Hanoi"
[...]
```
(I will email the maxmind  to make them know that error on ASN database. I take a look IP2Location and other ASN database, it is a common problem. )

Edited: Maxmind Geolite2 ASN database which releases on Mar 17 correct some error data:  https://gist.github.com/tomac4t/9c0c3139f6107731c926d3048cbecf4d

**Comment by sgiehl** [comment]
@tomac4t  I guess you're right. That piece of code maybe can be removed. Guess it was in there as if no database is used `gethostbyaddr` is used to get the "provider". But it is already cut up after the first dot within the detection. So guess there's no need to do that another time...
Guess that's something we can at least improve for Matomo 4
